packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts

```diff
@@ -597,6 +590,15 @@ export class GraphQLApi extends Construct {
     }
   }
 
+  /**
+   * Add schema dependency if not imported
+   * @param construct the construct that has a dependency
+   */
+  public addSchemaDependency( construct: CfnResource ): boolean {
+    construct.addDependsOn(this.schema);
+    return true;
```

Comment:
do we need this method? it seems like it's substituting one API call (`addDependsOn`) with another (`addSchemaDependency`). any reason users can't just do it directly?

does it require validation? what's valid as a construct to add

Reply:
there needs to be a dependency add for resolvers and the schema, but i was not comfortable passing schema's around in imports so essentially i needed a way to allow for adding a dependency only if the api wasn't imported. 

this basically was my workaround.. if anything i can make this function protected

Reply:
can  we make it internal?

Reply:
actually, I've thought about this a little more and I think that it might be good to offer a public api to add a dependency.. if the graphql api is in another stack, it will be deployed before the imported stack so the dependency doesn't matter too much? 

but in the case that it does we should have a public api to allow for adding dependencies? if not i dont see why a public api is a net bad

Reply:
if it enables use cases that users cannot achieve, then I agree.
this method calls `construct.addDependsOn(schema)` and returns true. can users just make that api call instead?

Reply:
@shivlaks its just really annoying for users to have to do after creating each resolver. 

The only reason its public as of right now is that I need `IGraphQLApi` to have some capability to reference a schema, I can't make it protected or the function will be inaccessible to the Resolver class.